src/features/organizations/rpc/getSuborgsWithStats.ts

```diff
@@ -0,0 +1,147 @@
+import { z } from 'zod';
+
+import IApiClient from 'core/api/client/IApiClient';
+import { makeRPCDef } from 'core/rpc/types';
+import {
+  ZetkinCampaign,
+  ZetkinEmail,
+  ZetkinEvent,
+  ZetkinEventParticipant,
+  ZetkinSmartSearchFilter,
+  ZetkinSubOrganization,
+  ZetkinSurveySubmission,
+  ZetkinView,
+} from 'utils/types/zetkin';
+import { ZetkinSmartSearchFilterStats } from 'features/smartSearch/types';
+import { FILTER_TYPE, OPERATION } from 'features/smartSearch/components/types';
+import { SuborgWithStats } from '../types';
+import { ZetkinCall } from 'features/call/types';
+import { ZetkinEmailStats } from 'features/emails/types';
+
+const paramsSchema = z.object({
+  orgId: z.number(),
+});
+
+type Params = z.infer<typeof paramsSchema>;
+
+export type Result = SuborgWithStats[];
+
+export const getSuborgsWithStatsDef = {
+  handler: handle,
+  name: 'getSuborgsWithStats',
+  schema: paramsSchema,
+};
+
+export default makeRPCDef<Params, Result>(getSuborgsWithStatsDef.name);
+
+async function handle(params: Params, apiClient: IApiClient): Promise<Result> {
+  const { orgId } = params;
+
+  const suborgs = await apiClient.get<ZetkinSubOrganization[]>(
+    `/api/orgs/${orgId}/sub_organizations`
+  );
+
+  const activeSuborgs = suborgs.filter((suborg) => suborg.is_active);
+
+  const now = new Date();
+  const thirtyDaysAgo = new Date(new Date().setDate(now.getDate() - 30))
+    .toISOString()
+    .slice(0, 10);
+  const today = now.toISOString().slice(0, 10);
+
+  const suborgPromises = activeSuborgs.map(async (suborg) => {
+    const [suborgStats, calls, surveySubmissions, events, lists, projects] =
+      await Promise.all([
+        apiClient.post<
+          ZetkinSmartSearchFilterStats[],
+          { filter_spec: ZetkinSmartSearchFilter[] }
+        >(`/api/orgs/${suborg.id}/people/queries/ephemeral/stats`, {
+          filter_spec: [
+            {
+              config: {},
+              op: OPERATION.ADD,
+              type: FILTER_TYPE.ALL,
+            },
+          ],
+        }),
+        apiClient.get<ZetkinCall[]>(`/api/orgs/${suborg.id}/calls?recursive`),
+        apiClient.get<ZetkinSurveySubmission[]>(
+          `/api/orgs/${suborg.id}/survey_submissions?recursive`
+        ),
+        apiClient.get<ZetkinEvent[]>(
+          `/api/orgs/${suborg.id}/actions?recursive&filter=start_time>=${thirtyDaysAgo}&filter=start_time<=${today}`
+        ),
+        apiClient.get<ZetkinView[]>(
+          `/api/orgs/${suborg.id}/people/views?recursive`
+        ),
+        apiClient.get<ZetkinCampaign[]>(
+          `/api/orgs/${suborg.id}/campaigns?recursive`
+        ),
```

Comment:
Right now you're looping through organizations and load their data recursively, like this:

* Load data for sub-org A and all of its sub-orgs
* Load data for sub-org B and all of its sub-orgs
* Load data for sub-org C and all of its sub-orgs
* …

I believe that you could instead just load make a _single recursive request_ (per type) using `orgId`:

* Load data for sub-org `orgId` and all of it's sub-orgs
  * Will include any events that may exist in A
  * Will include any events that may exist in B
  * Will include any events that may exist in C
  * …

That should be much more efficient.